compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -190,6 +191,10 @@ public boolean isAssignable(BType source, BType target) {
             return true;
         }
 
+        if (target.tag == TypeTags.TABLE && source.tag == TypeTags.TABLE) {
```

Comment:
Since SQL Connector action definitions returns a table without a constraint type (constrained get resolved in runtime), we cannot check this at semantic level. Adding a explicit cast  also not good as  sql action itself can return errors and cast also can return errors (which results in more line of codes to call sql select action)